.tests/alexa.rb

```diff
@@ -1,86 +1,96 @@
-#/usr/bin/ruby
-require "cgi"
-require "base64"
-require "openssl"
-require "uri"
-require "net/https"
-require "rexml/document"
-require "time"
-
-access_key_id = ENV["alexa_access_key"]
-secret_access_key = ENV["alexa_access_secret"]
-#
-# Sample request to Alexa Web Information Service
-#
-
-if ARGV.length < 1
-  $stderr.puts "Usage: urlinfo.rb site"
-  exit(-1)
-else
-  site = ARGV[0]
+#!/bin/ruby
+# frozen_string_literal: true
+
+require 'cgi'
+require 'base64'
+require 'openssl'
+require 'uri'
+require 'net/https'
+require 'rexml/document'
+require 'time'
+
+access_key_id = ENV['alexa_access_key']
+secret_access_key = ENV['alexa_access_secret']
+
+abort 'Usage: alexa.rb site' if ARGV.empty?
+
+abort 'API keys not set.' if access_key_id.nil? || secret_access_key.nil?
+
+if File.exist?("../tmp/#{ARGV[0]}.txt")
+  puts File.read("../tmp/#{ARGV[0]}.txt")
+  exit(0)
 end
 
-SERVICE_HOST = "awis.amazonaws.com"
-SERVICE_ENDPOINT = "awis.us-west-1.amazonaws.com"
+site = ARGV[0]
+SERVICE_HOST = 'awis.amazonaws.com'
+SERVICE_ENDPOINT = 'awis.us-west-1.amazonaws.com'
 SERVICE_PORT = 443
-SERVICE_URI = "/api"
-SERVICE_REGION = "us-west-1"
-SERVICE_NAME = "awis"
-
-def getSignatureKey(key, dateStamp, regionName, serviceName)
-  kDate    = OpenSSL::HMAC.digest('sha256', "AWS4" + key, dateStamp)
-  kRegion  = OpenSSL::HMAC.digest('sha256', kDate, regionName)
-  kService = OpenSSL::HMAC.digest('sha256', kRegion, serviceName)
-  kSigning = OpenSSL::HMAC.digest('sha256', kService, "aws4_request")
-  kSigning
+SERVICE_URI = '/api'
+SERVICE_REGION = 'us-west-1'
+SERVICE_NAME = 'awis'
+
+def getsignaturekey(key, datestamp, regionname, servicename)
+  kdate = OpenSSL::HMAC.digest('sha256', 'AWS4' + key, datestamp)
+  kregion = OpenSSL::HMAC.digest('sha256', kdate, regionname)
+  kservice = OpenSSL::HMAC.digest('sha256', kregion, servicename)
+  OpenSSL::HMAC.digest('sha256', kservice, 'aws4_request')
```

Comment:
Nitpick: usually ruby uses snake_case (makes things easier to read I think)
```suggestion
def get_signature_key(key, date_stamp, region_name, service_name)
  kdate = OpenSSL::HMAC.digest('sha256', 'AWS4' + key, date_stamp)
  kregion = OpenSSL::HMAC.digest('sha256', kdate, region_name)
  kservice = OpenSSL::HMAC.digest('sha256', kregion, service_name)
  OpenSSL::HMAC.digest('sha256', kservice, 'aws4_request')
```